src/content/React-CSharp/ClientApp/src/components/api-authorization/ApiAuthorizationRoutes.js

```diff
@@ -1,30 +1,23 @@
-import React, { Component, Fragment } from 'react';
+import React from 'react';
+import { Fragment } from 'react';
 import { Route } from 'react-router';
-import { Login } from './Login'
-import { Logout } from './Logout'
+import { Login } from './Login';
+import { Logout } from './Logout';
 import { ApplicationPaths, LoginActions, LogoutActions } from './ApiAuthorizationConstants';
 
-export default class ApiAuthorizationRoutes extends Component {
-
-  render () {
-    return(
-      <Fragment>
-          <Route path={ApplicationPaths.Login} render={() => loginAction(LoginActions.Login)} />
-          <Route path={ApplicationPaths.LoginFailed} render={() => loginAction(LoginActions.LoginFailed)} />
-          <Route path={ApplicationPaths.LoginCallback} render={() => loginAction(LoginActions.LoginCallback)} />
-          <Route path={ApplicationPaths.Profile} render={() => loginAction(LoginActions.Profile)} />
-          <Route path={ApplicationPaths.Register} render={() => loginAction(LoginActions.Register)} />
-          <Route path={ApplicationPaths.LogOut} render={() => logoutAction(LogoutActions.Logout)} />
-          <Route path={ApplicationPaths.LogOutCallback} render={() => logoutAction(LogoutActions.LogoutCallback)} />
-          <Route path={ApplicationPaths.LoggedOut} render={() => logoutAction(LogoutActions.LoggedOut)} />
-      </Fragment>);
-  }
-}
-
-function loginAction(name){
-    return (<Login action={name}></Login>);
-}
-
-function logoutAction(name) {
-    return (<Logout action={name}></Logout>);
+export function ApiAuthorizationRoutes() {
+  const loginAction = (name) => (<Login action={name}/>);
+  const logoutAction = (name) => (<Logout action={name}/>);
```

Comment:
nit: this functions should be declared outside of `ApiAuthorizationRoutes`

Reply:
I'm not sure that I fully understand the correct place for this functions. We already have separate `<Login />` and `<Logout />` components. This functions are only used to simplify possible changes in `ApiAuthorizationRoutes` and reduce code duplication. 

Is it correct to take them outside of `ApiAuthorizationRoutes` in the same file and what reasons can be used to this code formatting?

Reply:
generally i would try to avoid declaring functions inside components unless they close over some variables, but it's mostly down to preference